modules/vmss/main.tf

```diff
@@ -158,17 +158,34 @@ resource "azurerm_linux_virtual_machine_scale_set" "this" {
   }
 }
 
-resource "azurerm_application_insights" "this" {
-  count = var.metrics_retention_in_days != 0 ? 1 : 0
+resource "azurerm_log_analytics_workspace" "this" {
+  count = (length(var.autoscale_metrics) != 0 || try(var.app_insights_settings.create, false)) && try(var.app_insights_settings.workspace_mode, true) ? 1 : 0
```

Comment:
Is try(var.app_insights_settings.workspace_mode, true) should be try(var.app_insights_settings.workspace_mode, false) ?

Reply:
Same as the previous comment, the workspace is used by default.